ci/conda_env_sphinx.txt

```diff
@@ -26,6 +26,7 @@ pydata-sphinx-theme=0.14
 sphinx-autobuild
 sphinx-design
 sphinx-copybutton
+sphinx-lint
```

Comment:
Do we need this?
I think that `sphinx-lint` is installed by `pip install -e dev/archery[lint]`.

Reply:
Given that archery's setup.py has its own specific version pinning, it might indeed not be needed to include here

Reply:
I agree. Removed in b526a4d844b143b6ef5c414abf0366b274645817.